### ISSUE TITLE
LibreOffice won't start

### ISSUE THREAD

**Issue by jodumont** [pocl/pocl#662]
I know; I'm probably far from the libreoffice community but
when I run write, calc or even libreoffice
I receive this error

```
pocl error: lt_dlopen("(null)") or lt_dlsym() failed with 'file not found'.
note: missing symbols in the kernel binary might be reported as 'file not found' errors.
```
my kernel is: Linux znbk 4.18.0-kali1-amd64 #1 SMP Debian 4.18.6-1kali1 (2018-09-10) x86_64 GNU/Linux

I was working before like few month before, ..., hum I can't really track the change...

**Comment by franz** [comment]
The first thing to try is to remove pocl's cache. Can you try ``rm -rf $HOME/.cache/pocl`` and see if it helps ?

**Comment by jodumont** [comment]
:( no luck I have the same error even after a clean reinstall

**Comment by pjaaskel** [comment]
Does pocl work at all for you? That error message is usually correct, meaning that your kernel builtin lib is corrupted (e.g. refers to missing symbols). POCL_DEBUG=1 might help too.

**Comment by jodumont** [comment]
I also have the issue with ParrotOS which is also based on Debian Testing

Debian Testing have
└──╼ $sudo apt-get install libpocl2

which might be the issue no ?

**Comment by PalinuroSec** [comment]
Hello, i am a Parrot OS developer and i am experiencing the same issue.
Unfortunately libreoffice gave us a lot of issues with the sandbox.

I managed to have libreoffice working with the following patch, but i expect it to break again soon with the next firejail or libreoffice update (as usual)
https://nest.parrotsec.org/debian-packages/firejail/commit/c04610ece5ef91b12640bd0ce9d1f7dea0b4f537

**Comment by franz** [comment]
Hello,

I'm entirely unfamiliar with ParrotOS, and also what that patch does. I'm wildly assuming `noexec ${HOME}` somehow restricts executable files in the user's home directory ? That may interfere with pocl, because it compiles OpenCL source code to executable binaries, which are then loaded with `dlopen`. The default location where these are stored is `$HOME/.cache/pocl`, but can be changed by `POCL_CACHE_DIR=/to/some/path` env variable.

**Comment by PalinuroSec** [comment]
@franz thank you, how could we set up `POCL_CACHE_DIR` to have effect to a user? and is it safe to configure it to use `/tmp` ?

the patch changes the firejail sandbox profile for libreoffice to circumvent the problem, but we would really like to  keep the home folder secure without execution permissions

**Comment by franz** [comment]
@PalinuroSec yes `/tmp` should work, i use it all the time. You'll likely want something per-user, like `/tmp/pocl-<user-id>`, or perhaps `/run/user/<user-id>/pocl`. But any directory where you can write an executable library and dlopen it should work.

**Comment by pjaaskel** [comment]
Is this working now?
